lib/Runtime/Library/GlobalObject.cpp

```diff
@@ -599,10 +599,17 @@ namespace Js
         if (!scriptContext->IsInEvalMap(key, isIndirect, &pfuncScript))
         {
             uint32 grfscr = additionalGrfscr | fscrReturnExpression | fscrEval | fscrEvalCode | fscrGlobalCode;
+
             if (isLibraryCode)
             {
                 grfscr |= fscrIsLibraryCode;
             }
+
+            if (isIndirect)
+            {
+                grfscr |= fscrIndirect;
+            }
+
             pfuncScript = library->GetGlobalObject()->EvalHelper(scriptContext, argString->GetSz(), argString->GetLength(), moduleID,
```

Comment:
Updated in latest rev.

---
In reply to: [73385216](https://github.com/Microsoft/ChakraCore/pull/1377#discussion_r73385216) [](ancestors = 73385216)